### PR TITLE
Added platoons to the loadout system.

### PR DESCRIPTION
**Pull request by pool011**
**When merged this pull request will:**
- Add a platoon level to the loadout system to add depth to the system.
- Respect the [Development Guidelines](https://github.com/7Cav/cScripts/wiki/Code-and-development-policy)

### CODE REVIEW

File: cScripts/Loadouts/CfgLoadouts_Bravo_Atlas.hpp [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class Cav_B_Bravo_Atlas_base_F: Cav_B_Bravo_base_F {
     scope = 0;
+    platoon = 1;
```
Comment (AndreasBrostrom) [inline cScripts/Loadouts/CfgLoadouts_Bravo_Atlas.hpp:3]:
```suggestion
    platoon = 1;
    scope = 0;
```

File: cScripts/Loadouts/CfgLoadouts_Common.hpp [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class CommonBlufor {
     //regiment = "";
     //company = "";
+    //platoon = "";
```
Comment (AndreasBrostrom) [inline cScripts/Loadouts/CfgLoadouts_Common.hpp:4]:
```suggestion
    //platoon = "";
```
```suggestion
    //company = 0;
    //platoon = 0;
```

File: cScripts/Loadouts/CfgLoadouts_Common.hpp [hunk 3]
```diff
@@ -39,6 +41,7 @@ class Cav_B_Alpha_base_F: Cav_B_Seventh_Cavalry_Base_F {
 class Cav_B_Bravo_base_F: Cav_B_Seventh_Cavalry_Base_F {
     category[] = {"cScripts_Loadout_Cat_Bravo"};
     company = "bravo";
+    platoon = 0;
```
Comment (AndreasBrostrom) [inline cScripts/Loadouts/CfgLoadouts_Common.hpp:44]:
```suggestion
```

File: cScripts/Loadouts/CfgLoadouts_Common.hpp [hunk 4]
```diff
@@ -29,6 +30,7 @@ class Cav_B_Seventh_Cavalry_Base_F: CommonBlufor {
 class Cav_B_Alpha_base_F: Cav_B_Seventh_Cavalry_Base_F {
     category[] = {"cScripts_Loadout_Cat_Alpha"};
     company = "alpha";
+    platoon = 0;
```
Comment (AndreasBrostrom) [inline cScripts/Loadouts/CfgLoadouts_Common.hpp:33]:
```suggestion
```

File: cScripts/Loadouts/CfgLoadouts_Common.hpp [hunk 5]
```diff
@@ -49,6 +52,7 @@ class Cav_B_Bravo_base_F: Cav_B_Seventh_Cavalry_Base_F {
 class Cav_B_Charlie_base_F: Cav_B_Seventh_Cavalry_Base_F {
     category[] = {"cScripts_Loadout_Cat_Charlie"};
     company = "charlie";
+    platoon = 0;
```
Comment (AndreasBrostrom) [inline cScripts/Loadouts/CfgLoadouts_Common.hpp:55]:
```suggestion
```

**Comment by pool011** [comment]
Not ready to merge. Need to update getter/setter. Not working atm

**Comment by AndreasBrostrom** [comment]
# Wiki Article: Gear
cScripts Gear are our loadout script strongly inspired and taken from [Poppy](https://github.com/BaerMitUmlaut/Poppy/), our old loadout script. The gear system is rewritten and is greatly tailored to our community needs and have a extra simplification added to it. The loadout utilized a loadout array obtained from ace export loadout or the bis [`getUnitLoadout`](https://community.bistudio.com/wiki/getUnitLoadout) script command. The config also for customizing and can handle abilities and unit cosmetics such as rank and insignia.

### Loadouts
Loadouts are automatically applied to units on spawn based on the variable, variableName, classname or side. Loadouts are also saved when altered (*See below*). Loadouts using `Public` scope (*[See below](https://github.com/7Cav/cScripts/wiki/Gear#how-to-gear-up)*) in the config will also be present in loadout selectors such as ACE Arsenal default loadouts and [[staging]] and [[starter crate]] selectors.

### Loadout Saving
When you alter your loadout it automatically get saved and reapplied on respawn. The save event is run when you alter the loadout in **arsenal** or when you **close your inventory** and only happens when your inside of a defined [[Staging Zone|staging zone]]. All gear related functions will honer the saved loadout. Permissions are also kept based on the core loadout used before the save was committed.

## How to gear up
The gear system is quite simple it utilized config values to obtain and apply your loadout.
Loadouts are defined in the `CfgLoadouts.hpp`. The system allow loadout application for any side and unit and are inherited from parent classnames.
In order to make loadout for a side can setup a `Common<Blufor|Opfor|Independant|Civilian>` baseclass an example for Bluefor side can be found below.

```cpp
class CommonBlufor {};
class My_Loadout : CommonBlufor {
    regiment = "";
    company = "";
    platoon = -1;
    squad = -1;
    team = "";

    displayName = "";
    scope = 0;
    category[] = {};
    loadout = [[],[],[],[],[],[],"","",[],["","","","","",""]];
    equipmentTags[] = {};

    icon = "";
    insignia = "";

    abilityMedic = 0;
    abilityEngineer = 0;
    abilityEOD = 0;

    preLoadout = "";
    postLoadout = "";
};
```
### Config Properties
```cpp
class CommonBlufor {};
class My_Loadout : CommonBlufor {
CommonBlufor {
    regiment = "";          // Style value
    company = "";           // Used for arsenal whitelist based on side
    platoon = -1;           // Style value
    squad = -1;             // Used to determine squad name
    team = "";              // Used for auto coloring of fire team

    displayName = "";       // Used to make the selection label
    scope = 0;              // 0 private: not obtainable. 1 protected: can be used but not selected. 2 public: can be selected and used
    category[] = {};        // Hardcoded category paths found in ".\cScripts\CavFnc\functions\systems\fn_setupLoadoutCategories.sqf"
    loadout = [[],[],[],[],[],[],"","",[],["","","","","",""]]; // loadout array none quoted warped
    equipmentTags[] = {};   // Equipment tags

    icon = "";              // Style value (use classname of a unit icon)
    insignia = "";          // Style value (use classname of a insignia)

    abilityMedic = 0;       // 0 Nothing, 1 Medic, 2 Doctor
    abilityEngineer = 0;    // 0 Nothing, 1 Repair Specialists, 2 Engineer
    abilityEOD = 0;         // 0 Nothing, 1 EOD Specialist

    preLoadout = "";        // code that gets applied befor loadout and abilities are applied
    postLoadout = "";       // code that gets applied after loadout and abilities are applied
};
```

## Loadout applications
Loadouts can be applied in several ways. The main system used is the **classname** based system but it can also apply to units using **variableName**, **setVariable** or **side**. The loadout also need to be `public` or `protected` scope to be automatically applied, `public` to be select it via a selector.
```cpp
this setVariable ["cScripts_Gear_LoadoutClass", "Cav_B_C_Rifleman_F"];
```
### Keep spawn Loadout
You can keep the EDEN selected loadout by adding the following variable to a player unit:
```cpp
this setVariable ["cScripts_Gear_LoadoutClass", "default"];
```
_In our current setup of our loadouts `default` is not defined. This means it will not be applied. A warning of this will be shown in Eden SP and Eden MP._

## Arsenal
The arsenal is built automatically when opening the staging variant of the arsenal and utilizes side, company, weapon system and equipmentTags to construct the whitelist.

### Equipment Tags 
Equipment tags are defined in `CfgEquipmentTags.hpp` and called from a loadout in the `equipmentTags[]` array in `CfgLoadouts.hpp`. The equipment tags similar to the loadout system have side bases tags `Common<Blufor|Opfor|Independant|Civilian>` as well for company. Companies are defined as `Common<Blufor|Opfor|Independant|Civilian>Company<CompanyName>`. It is also possible to decare your own equipmentTags by adding a new config entry.
There are also one global equipment tag that is always applied to all whitelists named `Common`.

```
    Common[] = {"G_Aviator"};
    CommonBlufor[] = {
        "rhs_weap_mk18_KAC_bk",
        "rhs_weap_mk18_KAC",
        "rhs_mag_30Rnd_556x45_M855A1_PMAG_Tracer_Red"
    };
    CommonBluforCompanyCharlie[] = {"ace_banana"};
```

## See Also
- [[Staging Zone]]
- [[Starter Crate]]

**Comment by pool011** [comment]
This should be good to go

**Comment by pool011** [comment]
If we merge equipment tags then I think we should be ok. But @AndreasBrostrom I'll leave that up to your judgment.

**Comment by AndreasBrostrom** [comment]
thinking this as a in the middle. Will integrate this into the whitelist as well. but make it as seen in current state side and company automated.

**Pull request merged**